src/webots/nodes/WbGps.cpp

```diff
@@ -278,22 +278,6 @@ bool WbGps::refreshSensorIfNeeded() {
   for (int i = 0; i < 3; ++i)  // get exact position
     mMeasuredPosition[i] = t[i];
 
-  // if we are using 'WGS84' coordinate system with non-default 'northDirection'
-  // we need to adapt the exact position
-  if (WbWorld::instance()->worldInfo()->gpsCoordinateSystem().compare("WGS84") == 0) {
-    WbVector3 northDirection = WbWorld::instance()->worldInfo()->northDirection().normalized();
-    if (northDirection != WbVector3(1.0, 0.0, 0.0)) {
-      WbVector3 axis = northDirection.cross(WbVector3(1.0, 0.0, 0.0));
-      if (axis.isNull())
-        axis = WbVector3(0.0, 1.0, 0.0);
-      else
-        axis.normalize();
-      double angle = -northDirection.angle(WbVector3(1.0, 0.0, 0.0));
-      WbMatrix3 transformation = WbMatrix3(axis, angle);
-      mMeasuredPosition = mMeasuredPosition * transformation;
-    }
-  }
```

Comment:
We should make sure that the world imported from OSM are correct in term of latitude-longitude as this was the purpose of these lines.

Reply:
Yes, this is now done [here](https://github.com/cyberbotics/webots/blob/6345099d98f1967a4eeaedba1aa8cd03484a6ee0/src/webots/nodes/WbGps.cpp#L305-L311).